horace_core/sqw/PixelData/@PixelData/PixelData.m

```diff
@@ -554,19 +666,44 @@
             pix_idx_end = obj.num_pixels;
         end
 
-        obj.data = obj.f_accessor_.get_pix(pix_idx_start, pix_idx_end);
-        if obj.page_size == obj.num_pixels && obj.f_accessor_.is_activated()
-            % Close the file if all pixels have been read
-            obj.f_accessor_.deactivate();
+        obj.data_ = obj.f_accessor_.get_pix(pix_idx_start, pix_idx_end);
+        if obj.page_size == obj.num_pixels
+            % Delete accessor and close the file if all pixels have been read
+            obj.f_accessor_ = [];
         end
-        obj.pix_position_ = pix_idx_start;
     end
 
-    function obj = load_first_page_if_data_empty_(obj)
-        % Check if there's any data in the current page and load a page if not
-        if isempty(obj.data_) && ~isempty(obj.f_accessor_)
-            obj = obj.load_page_(1);
+    function obj = load_dirty_page_(obj, page_number)
+        % Load a page of data from a tmp file
+        raw_pix = obj.tmp_io_handler.load_page(page_number);
+        npix_cols = obj.PIXEL_BLOCK_COLS_;
+        obj.data_ = reshape(raw_pix, [npix_cols, numel(raw_pix)/npix_cols]);
+    end
+
+    function obj = write_dirty_page_(obj)
+        % Write the current page's pixels to a tmp file
+        obj.tmp_io_handler.write_page(obj.page_number_, obj.data);
+    end
+
+    function is = page_is_dirty_(obj, page_number)
+        % Return true if the given page is dirty
+        is = ~(page_number > numel(obj.page_dirty_));
+        is = is && obj.page_dirty_(page_number);
+    end
+
+    function obj = set_page_dirty_(obj, is_dirty, page_number)
+        % Mark the given page as "dirty" i.e. the data in the cache does not
+        % match the data in the original SQW file
+        %
+        % Input
+        % -----
+        % is_dirty     Logical specifying if the page is dirty
+        % page_number  The page number to mark as dirty (default is current page)
+        %
+        if nargin == 2
```

Comment:
The `nargin==2` check catches an omitted page number. What happens if `nargin==1`, i.e. the `is_dirty` flag is omitted?

Reply:
You get a `Not enough input arguments.` error